src/components/heading/heading.pw.tsx

```diff
@@ -0,0 +1,363 @@
+    test(`should render with ${characterVals} passed as subtitle`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponent subheader={characterVals} />);
+
+      await expect(subheaderPreview(page)).toHaveText(characterVals);
+    });
+  });
+
+  specialCharacters.forEach((characterVals) => {
+    test(`should render with ${characterVals} passed as subtitleId`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponent subtitleId={characterVals} />);
+
+      await expect(subheaderPreview(page)).toHaveId(characterVals);
+    });
+  });
+
+  specialCharacters.forEach((characterVals) => {
+    test(`should render with ${characterVals} passed as help text`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponent help={characterVals} />);
+
+      await headingHelp(page).hover();
+
+      await expect(headingHelpTooltip(page)).toBeVisible();
+      await expect(headingHelpTooltip(page)).toContainText(characterVals);
+    });
+  });
+
+  testData.forEach((helpLink) => {
+    test(`should render with ${testData} passed as help link`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponent helpLink={helpLink} />);
+
+      await expect(headingHelp(page)).toHaveAttribute("href", helpLink);
+    });
+  });
+
+  testData.forEach((backLink) => {
+    test(`should render with ${backLink} passed as back link`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponent backLink={backLink} />);
+
+      await expect(link(page).locator("a")).toHaveAttribute("href", backLink);
+    });
+  });
+
+  specialCharacters.forEach((characterVals) => {
+    test(`should render with ${characterVals} passed as helpAriaLabel`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <HeadingComponent
+          help="this is a title"
+          helpAriaLabel={characterVals}
+        />
+      );
+
+      await expect(headingHelp(page)).toHaveAttribute(
+        "aria-label",
+        characterVals
+      );
+    });
+  });
+
+  specialCharacters.forEach((characterVals) => {
+    test(`should render with pill and ${characterVals} passed as pill text`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponentWithPills pills={characterVals} />);
+
+      await expect(pillPreview(page)).toHaveText(characterVals);
+    });
+  });
+
+  dividerAndSeparatorValue.forEach(([dividerVals, renderState]) => {
+    test(`should render ${renderState} a divider when divider prop is ${dividerVals} `, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<HeadingComponent divider={dividerVals} />);
+
+      if (dividerVals) {
+        await expect(dividerPreview(page)).toHaveCSS(
+          "background",
+          "rgb(204, 214, 219) none repeat scroll 0% 0% / auto padding-box border-box"
+        );
+      } else {
+        expect(await dividerPreview(page).count()).toEqual(0);
+      }
```

Comment:
**suggestion(non-blocking)**: I would suggest any tests that include conditional assertions should ideally be split into two separate tests for each condition:

```tsx
it("should render a divider when divider prop is passed", () => { ... });

it("should not render a divider when divider prop isn't passed", () => { ... });
```


so its a bit clearer if tests start failing in the console what functionality isn't working correctly.